controller/api/destination/dst_overrides_adaptor.go

```diff
@@ -0,0 +1,51 @@
+package destination
+
+import (
+	"fmt"
+
+	"github.com/linkerd/linkerd2/controller/api/destination/watcher"
+	sp "github.com/linkerd/linkerd2/controller/gen/apis/serviceprofile/v1alpha2"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// dstOverridesAdaptor holds an underlying ProfileUpdateListener and updates
+// that listener with changes to the dstOverrides field of serviceProfiles.
+type dstOverridesAdaptor struct {
```

Comment:
I'm just curious, what do you think about changing the name of this to something like `defaultDstOverridesAdaptor`? To me, this adaptor feels like it provides a default value for `dstOverrides` if there are no other overrides from its parent publisher.

Reply:
Hmmm, That makes sense but I followed the same `opaqueports_adaptor` . Hence, not adding that extra prefix.

Another reason is that I'm also trying to move the authority port logic translator into the adaptor here (if It makes sense) 🤔